lib/widgets/channel_list.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/zulip_localizations.dart';
+
+import '../api/model/model.dart';
+import '../model/narrow.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'store.dart';
+
+class ChannelListPage extends StatelessWidget {
+  const ChannelListPage({super.key});
+
+  static Route<void> buildRoute({int? accountId, BuildContext? context}) {
+    return MaterialAccountWidgetRoute(accountId: accountId, context: context,
+      page: const ChannelListPage());
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final streams = store.streams.values.toList();
+    return Scaffold(
+      appBar: AppBar(title: Text(zulipLocalizations.channelListPageTitle)),
+      body: SafeArea(
+        child: ListView.builder(
+          itemCount: streams.length,
+          itemBuilder: (context, index) => ChannelItem(stream: streams[index]))));
+  }
+}
+
+@visibleForTesting
+class ChannelItem extends StatelessWidget {
+  const ChannelItem({super.key, required this.stream});
+
+  final ZulipStream stream;
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      color: Colors.white,
+      child: InkWell(
+        onTap: () => Navigator.push(context, MessageListPage.buildRoute(context: context,
+          narrow: ChannelNarrow(stream.streamId))),
+        child: Padding(
+          padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 16),
+          child: Row(children: [
+            Icon(size: 16, iconDataForStream(stream)),
+            const SizedBox(width: 8),
+            Expanded(child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(stream.name,
+                style: const TextStyle(
+                  fontSize: 18,
+                  height: (20 / 18),
+                  // TODO(#95) need dark-theme color
+                  color: Color(0xFF262626)),
+                maxLines: 1,
+                overflow: TextOverflow.ellipsis),
+                if (stream.description.isNotEmpty) Text(
+                  stream.description,
+                  style: const TextStyle(
+                    fontSize: 12,
+                    // TODO(#95) need dark-theme color
+                    color: Color(0xCC262626)),
```

Comment:
Do we want these colors go to one of the `ThemeExtension` classes we have? Maybe @chrisbobbe has an idea on that.

Reply:
Yeah, now that #95 is done, all new pieces of UI should support dark theme from the beginning. That means UI code gets colors from `DesignVariables` or another `ThemeExtension`, and colors for new UI get added there, with both light-theme and dark-theme values.